Added AuthorEntity

praveen4892 opened the pull request:
- Added required fields.

praveen4892 committed:
Added AuthorEntity

 - Added required fields.

SunnysGitAccount commented on src/main/java/com/dev/sunny/model/Author.java:
```diff
@@ -0,0 +1,18 @@
+import lombok.ToString;
+
+import java.util.List;
+@Getter
```
Add indentation

SunnysGitAccount commented on src/main/java/com/dev/sunny/model/Author.java:
```diff
@@ -0,0 +1,18 @@
+@Getter
+@Setter
+@ToString
+public class Author {
```
Add entity annotation

praveen4892 replied:
Added annotation

SunnysGitAccount commented on src/main/java/com/dev/sunny/model/Author.java:
```diff
@@ -0,0 +1,18 @@
+@Setter
+@ToString
+public class Author {
+
```
Add @Id, @GeneratedValue annotations

praveen4892 replied:
Added both

SunnysGitAccount reviewed (changes requested):
PR reviewed. There are some changes requested as the code change is incorrect and Author class is currently not an Entity.

praveen4892 committed:
Added requested changes

 - Added all the annotations required for AuthorEntity class.

SunnysGitAccount commented on src/main/java/com/dev/sunny/model/Author.java:
```diff
@@ -0,0 +1,26 @@
+public class Author {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
```
Add access specifiers to every attribute.

praveen4892 replied:
Done

SunnysGitAccount commented on src/main/java/com/dev/sunny/model/Author.java:
```diff
@@ -0,0 +1,26 @@
+    String name;
+    String email;
+    List<String> publications;
+
```
Add constructor with all attribute

 - Should be without id attribute.
 - Use @NoArgsConstructor to provide default constructor.

praveen4892 replied:
Resolved above changes

SunnysGitAccount commented on src/main/java/com/dev/sunny/model/Author.java:
```diff
@@ -0,0 +1,26 @@
+    String name;
+    String email;
+    List<String> publications;
+
```
Add equals and hash code implementation
 
- Use IntelliJ Default implementation (Recommendation).
- Use only id as not null member.

praveen4892 replied:
Done

SunnysGitAccount reviewed (changes requested):
Please update the class with requested changes.

praveen4892 committed:
Added requested changes.

 - Adds parameterized constructor.
 - Adds NoArgs constructor.
 - Adds equals and hashcode methods.

SunnysGitAccount reviewed (approved):
Good job completing PR Review changes. All changes are approved.
